src/backend/api/meals.js

```diff
@@ -1,48 +1,107 @@
+// @ts-nocheck
 const express = require('express');
 const router = express.Router();
 const knex = require('../database');
 
-router.get('/', async (request, response) => {
+router.get('/', async (req, res) => {
+  let query = knex('meal');
+  if ('maxPrice' in req.query) {
+    query = query.where('price', '<', Number(req.query.maxPrice));
+  }
+
+  // Its does not work
+  if ('availableReservations' in req.query) {
+    query = query.join('reservation', () => {
+      this.on('meal.id', '=', 'reservation.meal_id').orOn(
+        'number_of_guests',
+        '<',
+        'meal.max_reservation'
```

Comment:
This query should compare the SUM(number_of_guests) and max_reservation from meals.